### ISSUE TITLE
[v2] Refactor for merging into upstream microsoft/vscode

### ISSUE THREAD

**Issue by frank-dspeed** [cdr/code-server#1027]
# Refactor
- The patch needs to get splitted into smaller chunks so we can apply changes upstream 
- The build needs to get a bit refactored
- identify features not related to upstream
  - reuse existing marketplace code https://github.com/cdr/code-server/blob/master/src/marketplace.ts

Related Issues:
- #1022  https://github.com/cdr/code-server/issues/1021

# Upstream Epic
- https://github.com/microsoft/vscode/issues/81310
- Microsofts old implamentation WIP Current iteration in parallel https://github.com/microsoft/vscode/issues/74393

**Comment by sr229** [comment]
How exactly removing `marketplace.ts` solve anything? Did you test to see if it will work without it?

**Comment by frank-dspeed** [comment]
@sr229 current state is exploring how to refactor this repo and the vscode repo to produce a server build without patching so i need to find places where to put the code and don't double the code.

**Comment by frank-dspeed** [comment]
i think something like src/vs/platform/node-server and src/vs/workbench/workbench.server.api will do it but needs more information at present i am building multiple graphs of dependencys.

**Comment by sr229** [comment]
That would be really nice, we've been waiting for a specific moment to get rid of our patching and stuff. Please coordinate with @code-asher for this, he'll be down for getting the patches out in mainline

**Comment by frank-dspeed** [comment]
- Currently waiting for upstream

**Comment by nhooyr** [comment]
We can't merge into upstream?

**Comment by frank-dspeed** [comment]
@nhooyr why can't we do that ? this has working code it needs only a bit refactoring to be embeded it would be simply a extra vscode build and so also allow to use the nativ extension store.

**Comment by frank-dspeed** [comment]
https://github.com/microsoft/vscode/issues/81310

**Comment by nhooyr** [comment]
Microsoft already has their own version called vscode-web, they wouldn't want to merge ours.

**Comment by frank-dspeed** [comment]
@nhooyr they have done something total diffrent its called vscode-online by the way. there is still no backend in the code to run the web frontend.
